### DISCUSSION TITLE
is it possible to create Content driven elements

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2111] [Q&A]
### Introduction

```js
joint.shapes.newShapes.TextBox = joint.dia.Element.extend({
		defaults: _.defaultsDeep(
			{
				type: 'newShapes.TextBox ',
				size: { width: 100, height: 60 },
				attrs: {
					body: {
						refWidth: '100%',
						refHeight: '100%',
						strokeWidth: 2,
						stroke: 'red',
						fill: 'white',
					},
					label: {
						textWrap: {
							ellipsis: true,
						},
						textVerticalAnchor: anchorPosition.Middle,
						textAnchor: anchorPosition.Middle,
						refX: '50%',
						refY: '50%',
						fontSize: 14,
						fill: colorPalette.MediumGrey,
						text: 'Text',
					},
				},
			},
			joint.dia.Element.prototype.defaults,
		),

		markup: [
			{ tagName: 'rect', selector: 'body' },
			{ tagName: 'text', selector: 'label' },
		],
	});
```

I am using the above code to create new shapes, but in this element when I increase the size of the text it is disappearing 
I need to create dynamic elements 
eg: when I increase the text size automatically I need to increase the element size 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Here's a [tutorial](https://resources.jointjs.com/tutorial/content-driven-element) on the same.

Here are various [examples](https://www.jointjs.com/demos?feature=Content-driven+shapes).
